include/ensmallen_bits/adam/radam_update.hpp

```diff
@@ -0,0 +1,170 @@
+/**
+ * @file radam_update.hpp
+ * @author Shubham Agrawal
+ *
+ * RAdam update rule.
+ *
+ * ensmallen is free software; you may redistribute it and/or modify it under
+ * the terms of the 3-clause BSD license.  You should have received a copy of
+ * the 3-clause BSD license along with ensmallen.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#ifndef ENSMALLEN_ADAM_RADAM_UPDATE_HPP
+#define ENSMALLEN_ADAM_RADAM_UPDATE_HPP
+
+namespace ens {
+
+/**
+ * For more information, see the following.
+ *
+ * @code
+ * @article{Liu2019,
+ *   author  = {Liyuan Liu, Haoming Jiang, Pengcheng He, 
+ * Weizhu Chen, Xiaodong Liu, Jianfeng Gao, Jiawei Han},
+ *   title   = {On the Variance of the Adaptive Learning Rate and Beyond},
+ *   journal = {ICLR},
+ *   year    = {2019},
+ *   url     = {https://arxiv.org/abs/1908.03265}
+ * }
+ * @endcode
+ */
+class RAdamUpdate
+{
+ public:
+  /**
+   * Construct the Adam update policy with the given parameters.
+   *
+   * @param epsilon The epsilon value used to initialise the squared gradient
+   *        parameter.
+   * @param beta1 The smoothing parameter.
+   * @param beta2 The second moment coefficient.
+   */
+  RAdamUpdate(const double epsilon = 1e-8,
+             const double beta1 = 0.9,
+             const double beta2 = 0.999) :
+    epsilon(epsilon),
+    beta1(beta1),
+    beta2(beta2),
+    iteration(0),
+		pinf((2.0 / (1.0 - beta2)) - 1.0)
+  {
+    // Nothing to do.
+  }
+
+  //! Get the value used to initialise the squared gradient parameter.
+  double Epsilon() const { return epsilon; }
+  //! Modify the value used to initialise the squared gradient parameter.
+  double& Epsilon() { return epsilon; }
+
+  //! Get the smoothing parameter.
+  double Beta1() const { return beta1; }
+  //! Modify the smoothing parameter.
+  double& Beta1() { return beta1; }
+
+  //! Get the second moment coefficient.
+  double Beta2() const { return beta2; }
+  //! Modify the second moment coefficient.
+  double& Beta2() { return beta2; }
+
+  //! Get the current iteration number.
+  size_t Iteration() const { return iteration; }
+  //! Modify the current iteration number.
+  size_t& Iteration() { return iteration; }
+
+  /**
+   * The UpdatePolicyType policy classes must contain an internal 'Policy'
+   * template class with two template arguments: MatType and GradType.  This is
+   * instantiated at the start of the optimization, and holds parameters
+   * specific to an individual optimization.
+   */
+  template<typename MatType, typename GradType>
+  class Policy
+  {
+   public:
+    /**
+     * This constructor is called by the SGD Optimize() method before the start
+     * of the iteration update process.
+     *
+     * @param parent AdamUpdate object.
+     * @param rows Number of rows in the gradient matrix.
+     * @param cols Number of columns in the gradient matrix.
+     */
+    Policy(RAdamUpdate& parent, const size_t rows, const size_t cols) :
```

Comment:
For clarity, suggest to use `n_rows` and `n_cols` instead of `rows` and `cols`.  
This will also make it consistent with Armadillo: http://arma.sourceforge.net/docs.html#attributes

Reply:
For ensmallen we basically followed the mlpack style guidelines where we use `cols, rows` but I can see that this creates some confusion, so I don't mind to adapt the armadillo style here.

Reply:
I think we should stick with the mlpack style here, since ensmallen is part of the mlpack "umbrella".  It's good to have consistency across those projects (I know that this comes at the price of consistency with Armadillo).